### PR TITLE
complete the test

### PR DESCRIPTION
**Pull request by cieldon32**
I have write a version of SSR, the link : https://github.com/cieldon32/test-fullstack/tree/daily/0.0.1

### CODE REVIEW

**Comment by flyingant** [comment]
@cieldon32 Hi, this is MaYi from Wiredcraft. And I reviewed your code (mainly frontend).

Just be curious. 

I noticed that you have some code related to the Authentication flow in the file (https://github.com/Wiredcraft/test-fullstack/pull/42/files#diff-1970dda743901e6cf68f156ca293072e)

Are you able to explain what's the `Auth0Context` and how it works?

**Comment by cieldon32** [comment]
https://github.com/cieldon32/test-fullstack/blob/daily/0.0.2/web/src/index.provider.tsx

This file is copy from my other application. that will provide the global
state for the site.


but I think the test application not need, it . so I modify it , but not
cleaned up.


My auth strategy is
https://github.com/cieldon32/test-fullstack/blob/daily/0.0.2/server/src/auth/jwt.strategy.ts


Why do I let the server handle the web authentication?

because, I write the SSR version:
https://github.com/cieldon32/test-fullstack/blob/daily/0.0.1/src/auth/jwt.strategy.ts

And I read the user story. My understanding is that users who are not
logged in can see the talks list. So I let the post a poll need with a
token.

Liu Cheng a.k.a MaYi <notifications@github.com> 于2020年2月10日周一 下午10:00写道：

File: web/src/App.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import Layout from './components/layout';
+import Home from './containers/home';
+import Login from './containers/login';
+import EditPoll from './containers/editPoll';
+import Register from './containers/register';
+
+function App(): JSX.Element {
+
+  return (
+    <HashRouter>
```
Comment (haishanh) [inline web/src/App.tsx]: I see `BrowserRouter` used web/src/index.tsx

Why do we need `HashRouter` here?

**Comment by cieldon32** [comment]
This is a mistake. I should repeat ‘BrowserRouter’ to ‘HashRouter’.

Haishan <notifications@github.com> 于2020年2月11日周二 上午10:53写道：

File: server/.prettierrc [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "trailingComma": "all"
+}
```
Comment (haishanh) [inline server/.prettierrc]: why not add a similiar file in web/, the code style there is very inconsistent

**Comment by haishanh** [comment]
Thanks for the PR, sorry for nitpicking.

**Comment by cieldon32** [comment]
I have fixed these bugs.



best regards
carly xia


Haishan <notifications@github.com> 于2020年2月11日周二 下午5:35写道：

**Comment by makara** [comment]
Thanks. This should be enough for us to make a decision. We'll get back to you soon. Closing.

**Pull request closed without merging**